### PR TITLE
split: pass GNU tests/split/l-chunk.sh

### PR DESCRIPTION
**Pull request by zhitkoff**
Pass GNU  tests/split/l-chunk.sh test
Cleaning up tests

### CODE REVIEW

File: src/uu/split/src/split.rs [hunk 1]
```diff
@@ -1308,6 +1312,7 @@ where
 /// Where CHUNKS
 /// * l/N
 /// * l/K/N
+#[allow(clippy::cognitive_complexity)]
```
Comment (sylvestre) [inline src/uu/split/src/split.rs:1315]: instead of this, many you should split this function into smaller?

Comment (zhitkoff) [reply src/uu/split/src/split.rs:1315]: yeah, tried that originally, but ended up with a lot of duplicated code, hence the annotation. Will give anoter try and see if some code could be exctracted into "helper" functions

**Comment by sylvestre** [comment]
could you please rebase locally your PR and force push again ?
it says 82 commits here :)

**Comment by zhitkoff** [comment]
done

**Comment by tertsdiepraam** [comment]
I think the rebase didn't entirely go according to plan. It undoes some recent changes now.

**Pull request closed without merging**